### PR TITLE
Add migration guides for KV certs and keys

### LINKED ISSUE #12108: Create MigrationGuide.md for Key Vault
Create a migration guide for Key Vault similar to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide.md.

### PR DESCRIPTION
**Pull request by heaths**
Resolves #12108

### CODE REVIEW

**Comment by heaths** [comment]
Mainly just need to focus on the two MigrationGuide.md docs. The sample source is just to use our snippet generator to make sure it's valid code (I did this for secrets and caught a few "typos").

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/README.md:15]: Good idea!   I like this.

**Comment by heaths** [comment]
To those codeowners who just got added: I added links from your READMEs to your migration guides. I figured we should advertise those better, though not as prominently as API ref docs, samples (which, BTW, I noticed your READMEs don't have - I can add them now if you'd like) , etc. - hence putting them at the end.

**Review by AlexGhiondea — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/MigrationGuide.md
Comment (jsquire) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/MigrationGuide.md:23]: As I was reading this, I was going to comment that I wanted to steal and reuse it.  Then I realized why I liked it so much.  😛

**Review by jsquire — APPROVED** [review]
LGTM

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/MigrationGuide.md
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/MigrationGuide.md:23]: :) I "stole" it from Service Bus, though maybe the text is the same. @ramya-rao-a et. al. reformatted SB from Event Hubs. We probably want to do the same with EH. The side-by-side table doesn't render well, though the intent is certainly appreciated (e.g. I like side-by-side diffs better when doing reviews).

Comment (jsquire) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/MigrationGuide.md:23]: I agree with respect to the table, personally.   That was the format that was agreed upon across the languages for Event Hubs at the time.   I'll defer to @ramya-rao-a with respect as to whether we can either deviate per-language, coordinate cross-language on format updates, or leave as-is.

**Pull request merged**
